ga4gh/backend.py

```diff
@@ -217,6 +219,62 @@ def variantSetsGenerator(self, request):
         return self._topLevelObjectGenerator(
             request, self._variantSetIdMap, self._variantSetIds)
 
+    def readsGenerator(self, request):
+        # Local utility function to save some typing
+        def getPosition(readAlignment):
+            return readAlignment.alignment.position.position
+
+        if len(request.readGroupIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Read search over multiple readGroups not supported")
+        readGroupId = request.readGroupIds[0]
+        try:
+            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
+        except KeyError:
+            raise exceptions.ReadGroupNotFoundException(readGroupId)
+        startPosition = request.start
+        equalPositionsToSkip = 0
+        if request.pageToken is not None:
+            startPosition, equalPositionsToSkip = self.parsePageToken(
+                request.pageToken, 2)
```

Comment:
I believe I found a case where this algorithm fails, in that it could return reads outside the requested region.
Example:

Reads present on reference (with their ranges [start,end]):
`R1: [1, 100], R2: [2, 40], R3: [3, 100]`

Original client request:
Region: `[50,100]`, pagesize = `1`, no pageToken
 
Objects from `readGroup.getReadAlignment(…)`:
`R1, R3`
(R2 does not intersect requested range) so, first response is 
`[R1, pageToken=2:0]`

Next client request:
Region: `[50,100]`, pagesize = `1`, pageToken =` 2:0`

Objects from `readGroups.getReadAlignment(…)` (with startPosition reset from 50 to 2):
`R2, R3`
(R2 now intersects adjusted range), so second response is incorrect, contains spurious R2:
`[R2, pageToken=3:0]`

A solution would be to add an extra check in the first `while` loop below, to see if the read even intersects with the original requested region (substitute `readAlignment.end` here with whatever syntax is appropriate):
```
while getPosition(readAlignment) < startPosition or readAlignment.end < request.start:
    # etc...
```